GitUI/UserControls/BlameControl.cs

```diff
@@ -457,31 +457,54 @@ private void copyCommitHashToClipboardToolStripMenuItem_Click(object sender, Eve
             CopyToClipboard(c => c.ObjectId.ToString());
         }
 
+        private bool TryGetSelectedRevision(out GitRevision selectedRevision)
+        {
+            var blameCommit = GetBlameCommit();
+            if (blameCommit == null)
+            {
+                selectedRevision = null;
+                return false;
+            }
+
+            selectedRevision = _revGrid?.GetRevision(blameCommit.ObjectId);
+            return selectedRevision != null;
+        }
+
+        private void blameRevisionToolStripMenuItem_Click(object sender, EventArgs e)
+        {
+            if (!TryGetSelectedRevision(out var selectedRevision))
+            {
+                return;
+            }
+
+            BlameRevision(selectedRevision.ObjectId);
+        }
+
         private void blamePreviousRevisionToolStripMenuItem_Click(object sender, EventArgs e)
         {
-            int line = (int?)contextMenu.Tag ?? -1;
-            if (line < 0)
+            if (!TryGetSelectedRevision(out var selectedRevision))
             {
                 return;
             }
 
-            var objectId = _blame.Lines[line].Commit.ObjectId;
-            int originalLine = _blame.Lines[line].OriginLineNumber;
-            GitBlame blame = Module.Blame(_fileName, objectId + "^", _encoding, originalLine + ",+1");
-            if (blame.Lines.Count > 0)
+            var blameRevision = selectedRevision.HasParent
+                ? selectedRevision.FirstParentGuid
+                : selectedRevision.ObjectId;
+
+            BlameRevision(blameRevision);
+        }
+
+        private void BlameRevision(ObjectId revisionId)
+        {
+            if (_revGrid != null)
             {
-                var revision = blame.Lines[0].Commit.ObjectId;
-                if (_revGrid != null)
-                {
-                    _clickedBlameLine = blame.Lines[0];
-                    _revGrid.SetSelectedRevision(revision);
-                }
-                else
+                _revGrid.SetSelectedRevision(revisionId);
+            }
+            else
```

Comment:
nit: Our new coding style prefers early `return` over `else` in order to reduce nesting.